consensus/tendermint/tendermint.go

```diff
@@ -357,43 +291,26 @@ func q(totalVotingPower votingPower) votingPower {
 	return q
 }
 
-func (t *Tendermint[V, H, A]) handleFutureRoundMessage(header MessageHeader[A], addMessage func()) bool {
-	if header.Round > t.state.round {
-		if header.Round-t.state.round > maxFutureRound {
-			return false
-		}
-
-		addMessage()
-
-		if t.uponSkipRound(header.Round) {
-			t.doSkipRound(header.Round)
-		}
-		return false
+func (t *Tendermint[V, H, A]) preprocessMessage(header MessageHeader[A], addMessage func()) bool {
```

Comment:
This requires rewrite in order to move the `skipRound` rule outside of the function.